codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/OperationUnitTestGenerator.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.rust.codegen.client.smithy.generators
+
+import software.amazon.smithy.model.shapes.OperationShape
+import software.amazon.smithy.rust.codegen.client.smithy.ClientCodegenContext
+import software.amazon.smithy.rust.codegen.core.rustlang.RustWriter
+import software.amazon.smithy.rust.codegen.core.rustlang.rustTemplate
+import software.amazon.smithy.rust.codegen.core.rustlang.writable
+import software.amazon.smithy.rust.codegen.core.smithy.RuntimeType.Companion.preludeScope
+import software.amazon.smithy.rust.codegen.core.smithy.customize.writeCustomizations
+
+/**
+ * Generates operation-level unit tests
+ */
+class OperationUnitTestGenerator(
+    private val codegenContext: ClientCodegenContext,
+) {
+    fun render(
+        writer: RustWriter,
+        operationShape: OperationShape,
+        customizations: List<OperationCustomization>,
+    ) {
+        // TODO(smithy-rs#3759) don't generate this if there are no customizations adding unit tests.
+        writer.rustTemplate(
+            """
+            ##[cfg(test)]
+            mod test {
```

Comment:
This just randomly adds a `mod test` in the [middle of the operation files.](https://d2luzm2xt3nokh.cloudfront.net/codegen-diff/e4a58c360856bbbaa5c53e5db9ff25c1acaca0b9/14be1a5a285f614fb34b51d5541e1cd87c968a26/aws-sdk/index.html#file-0-0).

Can we not generate all of these into a dedicated module?

```rust
# lib.rs

#[cfg(smoketest)]
mod smoketests;
```

Reply:
Or as long as it's not worse than what procotol tests get rendered ([this](https://github.com/smithy-lang/smithy-rs/blob/main/aws/sdk/aws-models-extra/s3-tests.smithy#L172) gets rendered in the `put_object_test` mod in `put_object.rs`), maybe we are ok?

Reply:
These tests are now generated into their own file in the `tests` folder.